pkg/build/strategies/sti/sti.go

```diff
@@ -196,6 +205,15 @@ func (builder *STI) Prepare(config *api.Config) error {
 		WorkingDir: config.WorkingDir,
 	}
 
+	if len(config.RuntimeImage) > 0 {
+		// For pulling RuntimeImage currently we're using the same PullPolicy as for BuilderImage
+		_, err := dockerpkg.PullImage(config.RuntimeImage, builder.docker, config.BuilderPullPolicy, false)
```

Comment:
@wanghaoran1988 @bparees It's interesting idea but
1) I like the fail-fast approach
2) it looks very logical (at lest for me) to pull the image in `Prepare()` method
3) anyway you still have to pull the image. I wouldn't add async pull because it makes code less obvious but it will cover only this rare case (IMHO)

> i'll leave it up to @php-coder whether he thinks it would be too complicated to add that flow right now, it's definitely not a requirement

We can postpone it and implement later on demand.